controlplane/kubeadm/api/v1beta1/conversion.go

```diff
@@ -125,6 +131,14 @@ func Convert_v1beta1_KubeadmControlPlaneStatus_To_v1beta2_KubeadmControlPlaneSta
 	out.Deprecated.V1Beta1.UpdatedReplicas = in.UpdatedReplicas
 	out.Deprecated.V1Beta1.ReadyReplicas = in.ReadyReplicas
 	out.Deprecated.V1Beta1.UnavailableReplicas = in.UnavailableReplicas
+
+	// Move ready to Initialization
+	if in.Ready {
+		if out.Initialization == nil {
+			out.Initialization = &controlplanev1.KubeadmControlPlaneInitializationStatus{}
+		}
+		out.Initialization.ControlPlaneInitialized = in.Ready
+	}
```

Comment:
Shouldn't this be based on the v1beta1's Initialized?

```suggestion
	if in.Initialized {
		if out.Initialization == nil {
			out.Initialization = &controlplanev1.KubeadmControlPlaneInitializationStatus{}
		}
		out.Initialization.ControlPlaneInitialized = in.Initialized
	}
```